src/nat/tool/sequential_execution.py

```
@@ -0,0 +1,145 @@
+
+@register_function(config_type=SequentialExecutionToolConfig, framework_wrappers=[LLMFrameworkEnum.LANGCHAIN])
+async def sequential_execution(_config: SequentialExecutionToolConfig, builder: Builder):
+
+    tools: list[BaseTool] = builder.get_tools(tool_names=_config.sequential_tool_list,
+                                              wrapper_type=LLMFrameworkEnum.LANGCHAIN)
+    tools_dict = {tool.name: tool for tool in tools}
+    try:
+        input_type, output_type = _validate_sequential_tool_list(_config, builder)
+    except ValueError as e:
+        if _config.check_type_compatibility:
+            raise ValueError(f"The sequential tool list has incompatible types: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)